docs/rfcs/header-component-api.md

```diff
@@ -0,0 +1,469 @@
+
+#### Example
+
+```jsx
+<Subnav>
+    <Subnav.Link
+        label="UK"
+        href="https://www.theguardian.com/uk"
+    / >
+    <Subnav.Link
+        label="World"
+        href="https://www.theguardian.com/world"
+    / >
+<Subnav/>
+```
+
+#### `<Subnav.Link />`
+
+A navigational link that appears in the `Subnav` component
+
+![Subnav](images/header-component-api/subnav-link.png)
+
+##### Example
+
+```jsx
+<Subnav.Link label="UK politics" href="https://www.theguardian.com/politics" />
+```
+
+##### Props
+
+-   `href` _string_
+    -   navigation location when `Subnav.Link` is clicked
+-   `label` _string_
+    -   display label of the `Subnav.Link`
+
+## Specialised headers
+
+### `<EditorialHeader />`
+
+A special header layout for editorial pages. The logo is hard-coded.
+
+![Editorial header](images/header-component-api/editorial-header.png)
+
+#### Differences to generic header
+
+##### Masthead
+
+-   Links to the left, floated right
+-   Editions dropdown to the right
+-   My account / sign in
+    -   Links hierarchy
+    -   Primary link – always visible
+    -   Secondary links – visible at wider breakpoints
+    -   Editions menu
+-   Subscribe / Contribute links
+
+#### Example
+
+```jsx
+<EditorialHeader>
+    <Masthead>
+        <EditorialHeader.Support />
+        <Masthead.Links>
+            <Masthead.Link href icon={<Icon />} label until from />
+            <Masthead.Menu icon label until from>
+                <Masthead.Link href label />
+                <Masthead.Link href label />
+            </Masthead.Menu>
+            <Masthead.Link href icon={<Icon />} label until from />
+        </Masthead.Links>
+        <Masthead.Switch onChange>
+            <Masthead.SwitchItem href label />
+            <Masthead.SwitchItem href label selected />
+            <Masthead.SwitchItem href label />
+        </Masthead.Switch>
+    </Masthead>
+    <Navigation />
+    <Subnav />
+</EditorialHeader>
+```
+
+#### Props
+
+-   `edition` _string_ `'uk'` | `'au'` | `'us'` | `'international'`
+
+#### `<EditorialHeader.Support />`
+
+Presents either the Support buttons, or the Thank You message depending on whether the user is a supporter
+
+![Support buttons](images/header-component-api/support-buttons.png)
+![Support thank you message](images/header-component-api/support-thank-you.png)
+
+#### Props
+
+-   `showSupportButtons` _boolean_
+    -   whether to show the support buttons
+-   `showSubscriberMessage` _boolean_
+    -   whether to show the Thank You message
+-   `readerRevenueLinks` _{ subscribe: string; support: string; contribute: string }_
+    -   urls to assign to the support buttons
+
+## Risks and uncertainty
+
+-   The support buttons / thank you message is delivered by Automat
+    -   What is the state of Automat? Will it continue to be maintained, or should we take the opportunity to simplify?
+    -   Currently in support-dotcom-components. Should the component code live in Source?
```

Comment:
I'd be happy to chat about how these currently work and also some of the changes we are planning on making. As a really quick roundup:

The copy is configured (as A/B tests) through the Reader Revenue Control Panel (by our colleagues in BRR). The components are currently in support-dotcom-components (although we plan to move them to DCR repo in the future). When a user lands on a page, a request is sent to the support-dotcom-components backend which decides which A/B test & variant the user is in and then responds with the required data (mainly just copy). The user then makes a separate request for the component itself (although this is the part we plan to remove, and instead just have the components as a package in the DCR repo). 

I'm not quite sure how or if we'd want to have this baked into the source component? 

Happy to chat more!

Reply:
Hi @TomPretty thanks for the explanation. I'm a fan of reducing the complexity here, so thanks for your efforts 😍

Where these components ultimately live is up for grabs, but DCR is a good starting point. We're building the entire header here to begin with, so that aligns us nicely.

The copy – if there's already value in making that configurable, then we should replicate that in Source. I'll make changes to the API to show the intention. I'm guessing the current API can be found in the [header module folder](https://github.com/guardian/support-dotcom-components/tree/main/packages/modules/src/modules/header) in SDC?

Do you have a timeline for moving these components into DCR?